here-naksha-lib-psql/src/main/java/com/here/naksha/lib/psql/PostgresSession.java

```diff
@@ -621,10 +652,9 @@ private SQL prepareHstSql(
       //        partition_id = -1;
       //      }
       final PreparedStatement stmt = prepareStatement(
-          "SELECT r_op, r_id, r_uuid, r_type, r_ptype, r_feature, ST_AsEWKB(r_geometry), r_err\n"
-              + "FROM nk_write_features(?,?,?,?,?,?,?,?,?);");
-      // nk_write_features(col_id, part_id, ops, ids, uuids, features, geometries, min_result, errors_only
-      try (final Json json = Json.get()) {
+          "SELECT op, id, xyz, tags, feature, naksha_geometry(geo_type, geo), err_no, err_msg\n"
+              + "FROM naksha_write_features(?,?,?,?,?,?);");
+      try {
```

Comment:
Why is naksha_geometry used here? at least at the moment, we know that we only support three types, WKB, EWKB and TWKB, for all types we have readers. Converting to geometry will increase the size, because Postgres will convert TWKB into EWKB. This boils down to, deserializing into geometry, then serializing again without any use, why not just reading the type and bytes and read on java side?